include/gpiod.h

```diff
@@ -918,6 +918,28 @@ struct gpiod_line *
 gpiod_chip_get_line(struct gpiod_chip *chip, unsigned int offset) GPIOD_API;
 
 /**
+ * @brief Helper structure for storing a set of GPIO chip bulk objects.
+ */
+struct gpiod_chip_bulk {
+	struct gpiod_chip **chips;
+	/**< Dynamically allocated array of chip pointers. */
+	unsigned int num_chips;
+	/**< Number of chips currently held in this structure. */
+};
+
+/**
+ * @brief Open all available chips and return them in a bulk object
+ * @return the newly created GPIO chip bulk object, NULL on error
+ */
+struct gpiod_chip_bulk * gpiod_chip_open_all(void) GPIOD_API;
```

Comment:
The internals of this structure are known - let's not allocate it entirely, but just fill in the fields.